docs/architecture/README.md

```diff
@@ -3,111 +3,184 @@ order: 3
+- PostgreSQL
+- MySQL
+
+All server components will communicate through the database.
+
+::: warning
+When using a database, you can't scale the replica count for the scheduler, and you must have only **one** instance of it for the whole cluster.
+:::
+
+Database configuration options are available [here](../administrator-guide/configuration/databases/README.md).
+
+In this deployment mode, unless all components runs on the same host, you must use a distributed storage implementation like Google Cloud Storage, AWS S3 or Azure Blobs.
+
+### High availability with no single point of failure architecture
+
+![Kestra High Availability Architecture](./architecture.svg "Kestra High Availability Architecture")
+
+In order to support higher throughput, and full horizontal and vertical scaling of the Kestra cluster, we can replace the database with Kafka and Elasticsearch. In this case, all the server components can be scaled without any single point of failure.
+
+Kafka and Elasticsearch are available only in the **Enterprise Edition**.
+
+In this deployment mode, unless all components runs on the same host, you must use a distributed storage implementation like Google Cloud Storage, AWS S3 or Azure Blobs.
+
+#### Kafka
+
+[Kafka](https://kafka.apache.org/) is Kestra's primary dependency in high availability mode. Each of the most important server components in the deployment need to have a Kafka instance up and running. Kafka allows Kestra to be an highly scalable solution.
+
+Kafka configuration options are available [here](../administrator-guide/configuration/kafka/README.md).
+
+##### Kafka Executor
+
+With Kafka, the Executor is a heavy [Kafka Stream](https://kafka.apache.org/documentation/streams/) application. The Executor processes all events coming from Kafka in the right order, keeps an internal state of the execution, and merges task run results coming from the Worker.
+It also detects dead Workers and resubmits the tasks ran by a dead Worker.
+
+As the Executor is a Kafka Stream, it can be scaled as needed (within the limits of partitions count on Kafka), but as no heavy computations are done in the Executor, this server component does not require a lot of resources (unless you have a very high rate of executions).
+
+##### Kafka Worker
+
+With Kafka, the Worker is a [Kafka Consumer](https://kafka.apache.org/documentation/#consumerapi) that will process any Task Run submitted to it. Workers will receive all tasks and dispatch them internally in their Thread Pool.
+
+It can be scaled as needed (within the limits of partitions count on Kafka), and have many instances on multiple servers, each with its own Thread Pool. 
```

Comment:
```suggestion
It can be scaled as needed (within the limits of partitions count on Kafka) and have many instances on multiple servers, each with its own Thread Pool. 
```